src/olmo_core/data/numpy_dataset.py

```diff
@@ -270,9 +271,9 @@ def map(
 
         with executor_class(max_workers=max_workers) as executor:
             path_to_future = {}
-            for path in paths:
+            for idx, path in enumerate(paths):
                 if path not in path_to_future:
-                    path_to_future[path] = executor.submit(func, path)
+                    path_to_future[path] = executor.submit(func, (path, idx))
```

Comment:
If passing the index with the path now then each `(path, idx)` pair will be unique, so we should probably process all paths and not skip duplicates?

Reply:
Yeah, ok I see that now. I was trying to maintain backwards compat here in the case a subclass wants to keep the functionality. I guess could parameterize whether to skip dupe paths in this map method if you think it's worth keeping?